gammapy/catalog/hawc.py

```diff
@@ -0,0 +1,174 @@
+    """One source from the HAWC 2FHL catalog.
+
+    Catalog is represented by `~gammapy.catalog.SourceCatalog2HWC`.
+    """
+
+    _source_name_key = 'source_name'
+    _source_index_key = 'catalog_row_index'
+
+    def __str__(self):
+        return self.info()
+
+    def info(self, info='all'):
+        """Summary info string.
+
+        Parameters
+        ----------
+        info : {'all', 'basic', 'position', 'spectrum}
+            Comma separated list of options
+        """
+
+        if info == 'all':
+            info = 'basic,position,spectrum'
+
+        ss = ''
+        ops = info.split(',')
+        if 'basic' in ops:
+            ss += self._info_basic()
+        if 'position' in ops:
+            ss += self._info_position()
+        if 'spectrum' in ops:
+            ss += self._info_spectrum()
+
+        return ss
+
+    def _info_basic(self):
+        """Print basic info."""
+        d = self.data
+        ss = '\n*** Basic info ***\n\n'
+        ss += 'Catalog row index (zero-based) : {}\n'.format(d['catalog_row_index'])
+        ss += '{:<15s} : {}\n'.format('Source name:', d['source_name'])
+
+        return ss
+
+    def _info_position(self):
+        """Print position info."""
+        d = self.data
+        ss = '\n*** Position info ***\n\n'
+
+        ss += 'Measurement:\n'
+        ss += '{:20s} : {:.3f}\n'.format('RA', d['ra'])
+        ss += '{:20s} : {:.3f}\n'.format('DEC', d['dec'])
+        ss += '{:20s} : {:.3f}\n'.format('GLON', d['glon'])
+        ss += '{:20s} : {:.3f}\n'.format('GLAT', d['glat'])
+        ss += '{:20s} : {:.3f}\n'.format('Position error', d['pos_err'])
+
+        return ss
+
+    def _info_spectrum(self):
+        """Print spectral info."""
+        d = self.data
+        ss = '\n*** Spectral info ***\n\n'
+
+        ss += 'Spectrum 1:\n'
+        ss += '{:20s} : {:.3} +- {:.3} {}\n'.format('Flux at 7 TeV', d['spec0_dnde'].value, \
+            d['spec0_dnde_err'].value, 'cm-2 s-1 TeV-1')
+        ss += '{:20s} : {:.3f} +- {:.3f}\n'.format('Spectral index', d['spec0_index'], d['spec0_index_err'])
+        ss += '{:20s} : {:1}\n\n'.format('Test radius', d['spec0_radius'])
+
+        if(np.isnan(d['spec1_dnde'])):
+            ss += 'No second spectrum available for this source'
+        else:
+            ss += 'Spectrum 2:\n'
+            ss += '{:20s} : {:.3} +- {:.3} {}\n'.format('Flux at 7 TeV', d['spec1_dnde'].value, \
+                d['spec1_dnde_err'].value, 'cm-2 s-1 TeV-1')
+            ss += '{:20s} : {:.3f} +- {:.3f}\n'.format('Spectral index', d['spec1_index'], d['spec1_index_err'])
+            ss += '{:20s} : {:1}'.format('Test radius', d['spec1_radius'])
+
+        return ss
+
+    # TODO: Add an @property?
+    def _n_spec(self):
+        d = self.data
+        N = 0
+        if(np.isnan(d['spec1_dnde'])):
+            N = 1
+        else:
+            N = 2
+
+        return N
+
+    @property
+    def spectral_model(self):
```

Comment:
It's usually considered bad (confusing / hard to deal with return value / error prone) to have functions / methods / properties that sometimes return one thing, and sometimes something else.
At the moment you sometimes return a `SpectralModel` object, and sometimes a list of two spectral models.

I left a few suggestions how to do it before, but maybe I think maybe always returning a list of spectral models would be good and you like that solution? So make an empty list at the top, and then append one or two `SpectralModel` objects and return that.
* The implementation should be improved to avoid code duplication (currently you have the same code to copy over parameters to a SpectralModel 3 times -> should be there only once). See my `make_2hwc.py` script for an example how to do that.
* The power-law amplitude should be assigned a quantity wiht the correct units. In the tests you can evaluate the model e.g. at 1 TEV or 7 TeV to confirm that it works and gives the expected result.
* The docstring needs some more info, i.e. explain that it's a list of 1 or 2 spectral models, and give a sentence what they are (consult the paper description if needed).